bridges/snowbridge/docs/v2.md

```diff
@@ -0,0 +1,308 @@
+AliasOrigin Ethereum/Alice     // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+	DepositAsset (KLT, 100) dest=Bob
+```
+
+```
+WithdrawAsset (KLT, 100)
+ReserveAssetDeposited(WETH, 2)
+PayFees (WETH, 1)
+SetAssetClaimer Ethereum/Alice   // derived from beneficiary
+AliasOrigin Ethereum/Alice       // derived from msg.sender
+InitiateAssetsTransfer asset=(KLT, 100) remoteFee=(WETH, 1) dest=KLT
+		DepositAsset (KLT, 100) dest=Bob
+		Transact Bob.hello()
+```
+
+## Step 4: KILT Receives XCM from AH
+
+The flowing XCM $x_1$ is received from BH on AH.
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Bob
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+DepositAsset (KLT, 100) dest=Bob
+```
+
+```
+*WithdrawAsset (KLT, 100)
+*ReserveAssetDeposited (WETH, 1)
+*PayFees (WETH, 1)
+*SetAssetClaimer Bob
+*AliasOrigin Ethereum/Alice   // origin preserved from AH
+DepositAsset (KLT, 100) dest=Bob
+Transact Bob.hello()          // executes with the origin from AH
+```
+
+# Relayer Rewards
+
+The tracking and disbursement of relayer rewards for both directions has been unified. Rewards in WETH are accumulated on BH but must be manually claimed to deposit them into an account.
+
+To claim, call following extrinsic, where $o$ is rewards account (origin), and $w$ is account on AH where the WETH will be paid out.
+
+$$
+\mathrm{claim}(o,w)
+$$
+
+For tax accounting purposes might be desirable that $o \neq w$.
+
+# Top-Up
+
+Top-up on the destination is viable to implement.
+
+## Origin Preservation
+
+Origins for transact will be preserved by use of the `AliasOrigin` instruction. This instruction will have the following rules that parachain runtimes will need to allow:
+
+1. `AliasOrigin` can behave like `DescendOrigin`. This is allowed because it respects the hierarchy of multi-locations and does not allow jumping up.
+    1. Example location `Ethereum` can alias into `Ethereum/Alice` because we are descending in origin and this essentially is how the `DescendOrigin` instruction works.
+2. `AliasOrigin` must allow AH to alias into bridged locations such as `{ parents: 2, interior: GlobalConsensus(Ethereum) }` so that AH can act as a proxy for the bridge on parachains.
+
+`AliasOrigin` will be inserted by every `InitiateAssetTransfer` instruction on the source parachain, populated with the contents of the origin register, essentially forwarding the origin of the source to the destination.
+
+RFCS:
+
+[https://github.com/polkadot-fellows/RFCs/pull/122](https://github.com/polkadot-fellows/RFCs/pull/122)
+
+[https://github.com/polkadot-fellows/RFCs/blob/main/text/0100-xcm-multi-type-asset-transfer.md](https://github.com/polkadot-fellows/RFCs/blob/main/text/0100-xcm-multi-type-asset-transfer.md)
+
+## Parachain Requirements
+
+1. Pallet-xcm.execute enabled.
+2. XCM payment dry run apis.
+3. Must accept WETH needed for fees.
```

Comment:
You could add a comment here that this is only temporarily mandatory, user agents can be improved in the future to inject `ExchangeAsset give=(WETH, 1) want=(KLT, 20)` in the message executing on AH.